### DISCUSSION TITLE
FormViewModel Validation for Individual Form Fields

### DISCUSSION THREAD

**Discussion by ghost** [FilledStacks/stacked#530] [Q&A] [answered]
I am learning Stacked Architecture and working with the form generator. I see that in the FormViewModel class there is a single showValidation property and a single validationMessage property, which means that you can only have one validation message showing using setValidationMessage "out of the box". What would be the best way to have validation for each individual form field displaying under each field?

Edit: What seems to work is something like this but let me know what I should be doing.

Edit 2: I also have the use case where I want one (or more) of the FormTextFields to be readonly, and set that as a result of a calculation relying on the other fields, like in a calculator app, but it seems that the automatically generated stuff does not have a setter. Edit 3: oh, looks like the answer to this is here: https://github.com/FilledStacks/stacked/issues/382

```
class AgePayFormViewModel extends FormViewModel {
  final _navigationService = locator<NavigationService>();

  bool _showValidationAge = false;
  bool get showValidationAge => _showValidationAge;

  String? _validationMessageAge;
  String? get validationMessageAge => _validationMessageAge;

  void setValidationMessageAge(String? value) {
    _validationMessageAge = value;
    _showValidationAge = _validationMessageAge?.isNotEmpty ?? false;
  }

  @override
  void setData(Map<String, dynamic> data) {
    setValidationMessageAge(null);
    super.setData(data);
  }

  @override
  void setFormStatus() {
    // do validation for each field

    if (ageValue?.isEmpty ?? false) {
      // set validation message for age
      setValidationMessageAge("Enter an age");
    }
  }

  // If the dev doesn't want realtime validation then they can
  // simply validate in the function that they'll use to submit the
  // data to the backend or db.

  Future? saveData() {
    // here we can run custom functionality to save to our api
  }

  Future? navigateSomewhere() {
    // possible navigation
  }
}

```

**Comment by FilledStacks** [comment, marked as answer]
This has been implemented and is [documented here ](https://stacked.filledstacks.com/docs/getting-started/form-basics#form-validation)
